### PR TITLE
feat: support Nuxt 3 in for `@nuxtjs/emotion`

### LINKED ISSUE #20: Nuxt 3 compatibility
Hi,

Thanks for building this module. I was looking if I could use it for Nuxt 3, but the hooks are changed and it feels like it will need a re-write. That's why I'm creating an issue and not a PR.

I managed to get Emotion working in Nuxt 3 by creating a `server plugin` under `sever/plugins`. 
Nitro does have a `render:html` hook which can be used to get the `html` out of the pages.

This is the code I used to get Emotion working using `@emotion/server` and `@emotion/css`.

```js
import { extractCritical } from '@emotion/server'
import { hydrate } from '@emotion/css'

export default defineNitroPlugin((nitroApp) => {
  nitroApp.hooks.hook('render:html', (html) => {
    const { ids, css } = extractCritical(html.body)

    html.head.push(`<style data-emotion="${ids.join(' ')}">${css}</style>`)
    hydrate(ids)
  })
})
```

I did run into another issue where the caching for Emotion was not the same cache that was used on the client side. Thanks to @danielroe for helping me out not to inline emotion in dev mode by adding the following code to the `modules` array option in the Nuxt config.

```
modules: [
    ...
    function (options, nuxt) {
      nuxt.hook('nitro:config', config => {
        if (nuxt.options.dev) {
          // Prevent inlining emotion (+ the crucial css cache!) in dev mode
          config.externals.external ||= []
          config.externals.external.push('@emotion/server')
        }
      })
    }
  ]
  ```
  
  I'm currently sure this isn't the best solution so I hope it is a good start for the migration to Nuxt 3 for this module.
  
  Thanks!

### PR DESCRIPTION
**Pull request by codebender828**
Hi @pi0!

Thanks a lot for the work done with Nuxt 3. This pull request introduces support for the `@nuxtjs/emotion` module for Nuxt. It mostly constitutes code changes and changes to the README. 

I would like to gather some feedback on whether to keep the plugin options for this iteration of the `@nuxtjs/emotion` module for Nuxt 3. In all the demo projects I have worked on, I have not had to use any options for both client-side rendering, and server-side rendering. However, there may be some context that I might have missed. So your feedback on how to proceed would be greatly appreciated.

## Nuxt 3 Maintenance Suggestions
- **Nuxt 2**: A little bit of work should be done in the migration of the current `master` branch to a new branch called `v0`. That way the repo on the `master` branch defaults to Nuxt 3 support. Nuxt 2 code and docs would move to the `v0` branch.
- **Version management**: Since this code introduces a breaking change the version should be bumped to `v1.0.0`.
- **Add `"#imports"` support** for modules projects created with the `nuxi` CLI. In this pull request I resorted to manually type the `defineNitroPlugin` function because I was unable to locate the `#imports` types in VSCode. 

Thanks a lot to @danielroe for the support with understanding the Modules authoring process, and I am happy to answer any more questions.

Closes #20

### CODE REVIEW

**Comment by carwack** [comment]
This could close #20 👏

File: package.json [hunk 1]
```diff
@@ -4,34 +4,40 @@
...
-    "jest": "latest",
-    "nuxt": "latest",
-    "siroc": "latest",
-    "standard-version": "latest"
+    "@nuxt/module-builder": "^0.2.1",
+    "@nuxt/schema": "^3.0.0",
+    "@nuxt/test-utils": "^3.0.0",
+    "@nuxtjs/eslint-config-typescript": "^12.0.0",
+    "eslint": "^8.31.0",
+    "nitropack": "^1.0.0",
```
Comment (pi0) [inline package.json:38]: Do we need nitro as a dev dependency?

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Followup https://github.com/nuxt-community/emotion-module/issues/22

File: package.json (hunk 1 above)
Comment (codebender828) [reply package.json:38]: I think this will be removed. Thanks. I needed it only for development

**Pull request merged**